packages/vats/src/distributeFees.js

```diff
@@ -0,0 +1,113 @@
+// @ts-check
+
+import { observeNotifier } from '@agoric/notifier';
+import { E } from '@agoric/eventual-send';
+import { natSafeMath } from '@agoric/zoe/src/contractSupport';
+import { amountMath } from '@agoric/ertp';
+import { Far } from '@agoric/marshal';
+
+const { add, subtract, multiply, floorDivide } = natSafeMath;
+
+// wrapper to take the treasury's creatorFacet, and make a function that will
+// request an invitation and return a promise for a payment.
+export function makeTreasuryFeeCollector(zoe, treasuryCreatorFacet) {
+  /** @type FeeCollector */
+  return Far('collectFees', {
+    collectFees: () => {
+      const invitation = E(treasuryCreatorFacet).makeCollectFeesInvitation();
+      const collectFeesSeat = zoe.offer(invitation, undefined, undefined);
```

Comment:
Maybe this also needs:
```suggestion
      const collectFeesSeat = E(zoe).offer(invitation, undefined, undefined);
```

Reply:
How did I not notice that?